op-chain-ops/cmd/celo-migrate/ancients.go

```diff
@@ -45,13 +46,17 @@ func migrateAncientsDb(ctx context.Context, oldDBPath, newDBPath string, batchSi
 	if err != nil {
 		return 0, 0, fmt.Errorf("failed to open old freezer: %w", err)
 	}
-	defer oldFreezer.Close()
+	defer func() {
+		err = errors.Join(err, oldFreezer.Close())
+	}()
 
 	newFreezer, err := NewChainFreezer(filepath.Join(newDBPath, "ancient"), "", false)
 	if err != nil {
 		return 0, 0, fmt.Errorf("failed to open new freezer: %w", err)
 	}
-	defer newFreezer.Close()
+	defer func() {
+		err = errors.Join(err, newFreezer.Close())
+	}()
```

Comment:
You'll need to add named return parameters with the `error` being  `err` in order to access `err` correctly in your defer.